src/idl_parser.cpp

```diff
@@ -1635,3 +1635,78 @@ Offset<reflection::Type> Type::Serialize(FlatBufferBuilder *builder) const {
 }
 
 }  // namespace flatbuffers
+
+
+extern "C" {
+  FLATBUFFERS_EXPORT flatbuffers::Parser* flatbuffers_parser_new()
+  {
+    flatbuffers::Parser *p = new flatbuffers::Parser();
+    return p;
+  }
+
+  FLATBUFFERS_EXPORT int flatbuffers_parser_parse(flatbuffers::Parser *parser, const char* source)
+  {
+    if (parser == nullptr)
+    {
+      return -1;
+    }
+
+    bool result = parser->Parse(source, nullptr, nullptr);
+    if (result)
+    {
+      return 0;
+    }
+    return -1;
+  }
+
+  FLATBUFFERS_EXPORT void flatbuffers_free_string(char **buffer)
+  {
+    if (buffer != nullptr)
+    {
+      free(buffer);
+    }
+  }
+
+  FLATBUFFERS_EXPORT int flatbuffers_generate_json(flatbuffers::Parser *parser, char **buffer, size_t *size)
+  {
+    std::string jsongen;
+    flatbuffers::GenerateText(*parser, parser->builder_.GetBufferPointer(), &jsongen);
+    *size = jsongen.length();
+
+    *buffer = static_cast<char*>(malloc(sizeof(char) * *size));
+    memcpy(*buffer, jsongen.c_str(), *size);
+    return *size;
+  }
+
+  FLATBUFFERS_EXPORT int flatbuffers_generate_buffer(flatbuffers::Parser *parser, char **buffer, size_t *size)
```

Comment:
It doesn't have clear reason. In my experience, some OSS library used this signature (out param and return type) so just use this.

Certainly, it makes no sense, i'll fix this later.